### ISSUE TITLE
pi0 on franka embodiment?

### ISSUE THREAD

**Issue by NakuraMino** [Physical-Intelligence/openpi#310]
![Image](https://github.com/user-attachments/assets/16c03ed0-b7d0-4cf0-a838-d69e1fedb721)

Hello, the pi0 paper shows pi0 was trained on a franka arm with 2 cameras. However, when I load the pi0 models, none of them quite match this specification. The closest is the pi0_droid, but this requires a wrist camera. What configuration should I use for pi0 model to use on a franka with 2 cameras? 

Thank you!
Mino

**Comment by kpertsch** [comment]
Hi, 

We just added some further details for how to reload normalization statistics for existing robots in our pre-training mixture: https://github.com/Physical-Intelligence/openpi/blob/main/docs/norm_stats.md

So if you want to use a Franka, you can follow the instructions there to set up your action space + proprio vector and try reloading the norm stats. For setting up the training configs for your Franka, you can follow the examples we added for UR5 and modify to fit your Franka dataset: https://github.com/Physical-Intelligence/openpi/blob/main/examples/ur5/README.md

**Comment by NakuraMino** [comment]
Hi @kpertsch , 

Thank you for the response. I appreciate the release on the norm statistics for adjusting the action space. 

Apologies if my first question was not clear, but I was more interested in matching the observation space. The UR5 example also uses a wrist-mounted and over-the-shoulder camera. However, my setup uses two over-the-shoulder cameras, similar to the franka setup mentioned in the paper. Is there a way to match the observation space to use two over-the-shoulder cameras with pi0?

**Comment by kpertsch** [comment]
Hi, 
Pi0 is currently trained with only a single external and up to two wrist cameras. If your Franka setup has two external cameras, you could try feeding the second external cam in place of one of the wrist cams (preferrably the first wrist slot). Transfer may be worse but the model should handle it.
Hope this answers the questions, will close the issue for now, but feel free to reopen if anything is still unclear.

**Comment by HongyuanChen** [comment]
Hi @kpertsch ，

I'm actually encountering a similar problem, but I'd like to confirm the details further. Currently, I'm using the Franka robot with a total of 3 cameras: 1 wrist camera and 2 external cameras. I checked the `aloha_policy` and found that padding is True as long as camera data exists. For Droid, the padding for `right_wrist_0_rgb` is False. For Libero (the policy with the most comments and the most explanation for our custom dataset), `right_wrist_0_rgb` is only True for `PI0_FAST`, and False for the rest.

Considering your explanation above, is the optimal implementation like this: `left_wrist_0_rgb` is the `wrist_camera` from my own data, one of the external cameras is used as `right_wrist_0_rgb`, and the padding for `right_wrist_0_rgb` is False?
